web/static/js/study-responses.js

```diff
@@ -222,18 +222,20 @@ const resp_table = $("#individualResponsesTable").DataTable({
 // Date Range UI and filter for "Date" column
 setupDataTableDates("individualResponsesTable", 3, "dateRangeFilter");
 
-// Toggle the filled/unfilled star image visibility on input checkbox state change
-$('.star-checkbox').change(function () {
-    $(this).labels().children('.icon-star').toggleClass('icon-hidden');
-});
-
-// Update the sort/filter data values on td
-function updateCellData(select) {
-    const td = select.parentElement;
-    const text = select.options[select.selectedIndex].text;
-
-    td.dataset.sort = text;
-    td.dataset.filter = text;
+function updateAJAXCellData(target) {
+    const classes = target.classList
+    const td = target.parentElement
+
+    if (classes.contains("dropdown-cell")) {
+        const text = target.options[target.selectedIndex].text;
+        td.dataset.sort = text;
+        td.dataset.filter = text;
+    } else if (classes.contains("star-checkbox")) {
+        td.querySelectorAll('.icon-star').forEach(el => {
+            el.classList.toggle('icon-hidden')
+        })
+        td.dataset.sort = "False" == td.dataset.sort ? "True" : "False"
+    }
```

Comment:
Just to clarify: this toggles the `td` dataset sort value (True -> False and False -> True) for the star checkbox after it's been changed/saved. Right?